ethereum/contracts/ParachainClient.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "./BeefyClient.sol";
+import "./utils/MerkleProof.sol";
+import "./ScaleCodec.sol";
+import "./utils/MMRProofVerification.sol";
+
+contract ParachainClient {
+    using ScaleCodec for uint32;
+
+    BeefyClient public immutable beefyClient;
+    uint32 public immutable parachainID;
```

Comment:
Having `parachainID` as immutable might be an issue if Parity decides to swap our para id. Maybe we should have a way to update this? Not really sure how that would effect validation process so maybe we need to stress that we need get a stable parachain id from them during the onboarding process?

Reply:
i think governance upgrading to migrate the channels to a new parachain client should be fine to cover this risk